git.html

```diff
@@ -0,0 +1,151 @@
+                </ul>
+            </section>
+            <section id="update">
+                <h2>Safest way to “update” a local copy <a href="#update">&#x1F517;</a></h2>
+                <p>I find this sequence of commands the “safest” way to quickly “refresh” a clone of some remote repo, and at the same time check its status
+                    (where “safest” means <em>“reducing to the minimum the probability of messing up things with conflicts, outdated branches, uncommitted
+                    changes, etc”</em>):</p>
+                <pre>$ git branch -a
+$ git pull -r
+$ git status
+$ git remote prune origin --dry</pre>
+                <ul><li><code>$ git branch -a</code> displays information about <em>all</em> branches, both local and <em>remote</em></li>
+                    <li><code>$ git pull -r</code> fetches changes from the <em>remote</em> and
+                        <a href="https://git-scm.com/docs/git-pull#git-pull--r">“rebases the current branch on top of the upstream branch after fetching”</a>
+                    </li>
+                    <li><code>$ git status</code> shows you the status of your copy: whether there are new files, missing files, unstaged changes, or commits
+                        pending push</li>
+                    <li><code>$ git remote prune origin --dry</code> tells you if any branch in the <em>remote</em> has been recently removed.
+                        (Submit the same command <em>without</em> the <code>--dry</code> part to actually remove those remotes from your local
+                        <code>origin</code>.
+                        That will still <em>not</em> remove local branches automatically, but you can do that yourself with
+                        <code>git branch -d &lt;BRANCH&gt;</code> if you see some branch is no longer necessary.)
+                    </li>
+                </ul>
+                <p>You can have those four lines as an <em>alias</em>, or inside a <em>script</em> somewhere.</p>
+                <p>Even better: if you set up <a href="#aliases">the aliases suggested above</a>, the whole thing becomes:</p>
+                <pre>$ git br -a;git pl -r;git st;git re prune origin --dry</pre>
+                <p>You can then type it and run it just once, and, from that moment on, simply recover the line from your shell history.</p>
+                <p>For example, if you're using Bash: press <code>Ctrl</code>+<code>r</code>, then start typing a distinctive chunk of the line
+                    (eg, <code>r;</code>, or <code>st;gi</code>); if you used it not too long ago for the last time, the entire line should appear, and you can
+                    simply press <code>Enter</code>.</p>
+            </section>
+            <section id="lg">
+                <h2>An alias to view the history of the repo <a href="#lg">&#x1F517;</a></h2>
```

Comment:
That alias to display logs in tree is useful as it provides something more that just saving a few keystrokes ! so +1 to document that one.